code/DocGenerator.php

```diff
@@ -20,19 +20,30 @@ class DocGenerator extends Object implements Flushable
     /**
      * After the request, generate the docs.
      *
-     * @inheritdoc
+     * {@inheritdoc}
      */
     public static function flush()
     {
-        if (Config::inst()->get('DocGenerator', 'enabled') === true) {
+        $director = Injector::inst()->get('Director');
+        if ((Config::inst()->get('DocGenerator', 'enabled') === true) && (Config::inst()->get('DocGenerator', 'multi_module_enabled') === true)) {
```

Comment:
I like the idea, but could you extract it to a separate method? That would make everything much more readable :)